src/Http/Controllers/Requests/DefaultFormRequest.php

```diff
@@ -61,6 +94,37 @@ private function prepareMedia(array $data): array
         return $data;
     }
 
+    /**
+     * @param array<string, mixed> $data
+     * @return void
+     */
+    private function clearTempMediaFiles(array $data): void
+    {
+        foreach ($data as $key => $value) {
+            if (! is_array($value)) {
+                continue;
+            }
+
+            if ($key !== '_media' || ! $this->isValidMediaFileArray($value)) {
+                $this->clearTempMediaFiles($value);
+
+                continue;
+            }
+
+            foreach ($value as $mediaFiles) {
+                foreach ($mediaFiles as $mediaFile) {
+                    if (! $mediaFile instanceof MediaFile) {
```

Comment:
At this point you already know it's a MediaFile right? Because of the `isValidMediaFileArray` call.